fetchai/ledger/transaction.py

```diff
@@ -1,17 +1,33 @@
 import random
 from collections import OrderedDict
-from typing import Union
+from typing import Union, List
 
 from .bitvector import BitVector
 from .crypto import Address, Identity
 
 Identifier = Union[Address, Identity]
 
 
+class Transfer:
+    def __init__(self, to: Identifier, amount: int):
+        # ensure the address is correct
+        self.to = Address(to)
+        self.amount = amount
+
+    def __eq__(self, other):
+        return isinstance(other, Transfer) and self.to == other.to and self.amount == other.amount
+
+    def __hash__(self):
+        return hash((self.to, self.amount))
+
+
+TransferList = List[Transfer]
+
+
 class Transaction:
     def __init__(self):
         self._from = None
-        self._transfers = OrderedDict()
+        self._transfers: TransferList = []
```

Comment:
Invalid syntax in Python3.5
```suggestion
        self._transfers = []
```